pkg/build/builder/common.go

```diff
@@ -82,3 +89,107 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds of the same originalTag.
+func randomBuildTag(originalTag string) string {
+	repo, _ := parsers.ParseRepositoryTag(originalTag)
+	suffix := fmt.Sprintf("%08x", rand.Uint32())
```

Comment:
I was thinking of kapi.SimpleNameGenerator, but annoyingly the type and params aren't exported, so it's unusable outside the assumed length and suffix. `kutilrand.String(randomlength)` will do what https://github.com/openshift/origin/pull/6715/files#r51114131 suggested.

Looking at how this is used, though, are we ok with creating new tags for every build? Wondering how that will interact with pruning